external-api/src/cronJobs/eStoreFacilityFolderCreationJob.cron.ts

```diff
@@ -7,53 +7,60 @@ import { createFacilityFolder, uploadSupportingDocuments } from '../v1/controlle
-        {
-          $set: {
-            'facilityCronJob.status': ESTORE_CRON_STATUS.COMPLETED,
-            'facilityCronJob.completionDate': new Date(),
-          },
-        },
+    if (response?.value?.facilityFolderRetries <= 3) {
+      console.info('Cron task started: Create the Facility folders %s', response?.value?.facilityFolderRetries);
+      // create the Facility folders
+      const facilityFoldersResponse: any = await Promise.all(
+        eStoreData.facilityIdentifiers.map((facilityIdentifier: number) =>
+          createFacilityFolder(eStoreData.siteId, eStoreData.dealIdentifier, {
+            exporterName: eStoreData.exporterName,
+            buyerName: eStoreData.buyerName,
+            facilityIdentifier: facilityIdentifier.toString(),
+          }),
+        ),
       );
 
-      // stop and the delete the cron job - this in order to release the memory
-      eStoreCronJobManager.deleteJob(`Facility${eStoreData.dealId}`);
+      if (facilityFoldersResponse.every((item: any) => item.status === 201)) {
+        console.info('Cron task completed: Facility folders have been successfully created');
+        // stop and the delete the cron job - this in order to release the memory
+        eStoreCronJobManager.deleteJob(`Facility${eStoreData.dealId}`);
 
-      // check if there are any supporting documents
-      if (eStoreData.supportingInformation.length) {
-        console.info('Task started: Upload the supporting documents');
-        const uploadDocuments = Promise.all(
-          eStoreData.supportingInformation.map((file: any) =>
-            uploadSupportingDocuments(eStoreData.siteName, eStoreData.dealIdentifier, eStoreData.buyerName, { ...file }),
-          ),
+        // update the record inside `cron-job-logs` collection to indicate that the cron job finished executing
+        await cronJobLogsCollection.updateOne(
+          { dealId: { $eq: eStoreData.dealId } },
+          {
+            $set: {
+              'facilityCronJob.status': ESTORE_CRON_STATUS.COMPLETED,
+              'facilityCronJob.completionDate': new Date(),
+            },
+          },
         );
-        uploadDocuments.then((response) => console.info('Task completed: Supporting documents uploaded successfully', response[0].data));
-        uploadDocuments.catch((e) => console.error('Task failed: There was a problem uploading the documents', { e }));
+
+        // check if there are any supporting documents
+        if (eStoreData.supportingInformation.length) {
+          console.info('Task started: Upload the supporting documents');
+          const uploadDocuments = Promise.all(
+            eStoreData.supportingInformation.map((file: any) =>
+              uploadSupportingDocuments(eStoreData.siteId, eStoreData.dealIdentifier, eStoreData.buyerName, { ...file }),
+            ),
+          );
+          uploadDocuments.then((res) => console.info('Task completed: Supporting documents uploaded successfully', res[0].data));
```

Comment:
Can we please use `%s` string substitution for first and `%O` for second please?